src/main/scala/fi/oph/koski/valpas/hakukooste/SureHakukoosteService.scala

```diff
@@ -19,8 +22,15 @@ class SureHakukoosteService(config: Config) extends ValpasHakukoosteService with
     (oppijaOids: Set[ValpasHenkilö.Oid], ainoastaanAktiivisetHaut: Boolean = false, errorClue: String = "")
   : Either[HttpStatus, Seq[Hakukooste]] = {
     val encoder = json4sEncoderOf[Seq[ValpasHenkilö.Oid]]
-    val decoder = parseJson[Seq[Hakukooste]] _
 
+    def validate(parsedJson: JValue, request: Request): Seq[Hakukooste] = {
+      validatingAndResolvingExtractor.extract[Seq[Hakukooste]](parsedJson) match {
+        case Left(status) => throw HttpStatusException(status.statusCode, status.errors.map(_.toString).mkString(", "), request)
+        case Right(value) => value
+      }
+    }
```

Comment:
Tulee mieleen tarvitaanko tätä metodia lainkaan vai voisiko validaatiotuloksen käsitellä sellaisenaan Eitherinä?

Eli ehdottaisin että `parseJsonWithValidate`-funktio ottaisi parametrina `ValidatingAndResolvingExtractor`-instanssin ja palauttaisi `Either[HttpStatus, T]`. Tämän jälkeen validointivirheen voisi käsitellä tavalliseen tapaan.

Reply:
Toteutinkin itse asiassa aluksi näin, mutta sitten totesin, että geneerisen `Http`-luokan ei mielestäni kuulu olla riippuvainen Kosken sisäisestä datan sisällön validoinnista `ValidatingAndResolvingExtractor`-luokan kautta. Http-luokassa on myös käytetty keskeytyksiä Either:in sijasta, joten sen sisäinen konsistenssi virheiden käsittelyssä olisi mielestäni vähän seonnut siitä, että yhdessä metodissa virheitä käsitelläänkin eri tavalla.

Reply:
Tämä tehty!